train_cnn_lstm_enhanced.py

```
@@ -0,0 +1,831 @@
+)
+from trading_rl_agent.models.cnn_lstm import CNNLSTMModel
+
+logger = logging.getLogger(__name__)
+
+
+class EnhancedCNNLSTMTrainer:
+    """Enhanced trainer for CNN+LSTM models with comprehensive monitoring and optimization."""
+
+    def __init__(
+        self,
+        model_config: Dict[str, Any],
```

Ruff (0.11.9)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)